specification/cognitiveservices/data-plane/Language/preview/2021-11-01-preview/analyzeconversations.json

```
@@ -495,19 +493,683 @@
+          "x-ms-enum": {
+            "modelAsString": false
+          }
+        }
+      },
+      "required": [
+        "extraInformationKind"
+      ]
+    },
+    "EntitySubtype": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseExtraInformation"
+        }
+      ],
+      "type":"object",
+      "x-ms-discriminator-value": "EntitySubtype",
+      "properties": {
+        "value": {
+          "type":"string",
+          "description": "The Subtype of an extracted entity type."
+        }
+      }
+    },
+    "ListKeys":{
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseExtraInformation"
+        }
+      ],
+      "type":"object",
+      "x-ms-discriminator-value": "ListKeys",
+      "properties": {
+        "keys": {
+          "type": "array",
+          "description": "A Collection of keys that the extracted entity was mapped to.",
           "items": {
             "type": "string"
           }
         }
       }
     },
+    "BaseResolution": {
+      "type":"object",
+      "discriminator": "resolutionKind",
+      "properties": {
+        "resolutionKind":{
+          "type": "string",
+          "enum": [
+            "Boolean",
+            "DateTime",
+            "Number",
+            "Ordinal",
+            "Speed",
+            "Weight",
+            "Length",
+            "Volume",
+            "Area",
+            "Age",
+            "Information",
+            "Temperature",
+            "Currency",
+            "NumericRange",
+            "TemporalSpan"
+          ],
+          "x-ms-enum": {
+            "name": "ResolutionKind",
+            "modelAsString": false
+          }
+        }
+      },
+      "required": [
+        "resolutionKind"
+      ]
+    },
+    "QuantityResolution":{
+      "description": "represents resolutions for quantities.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "number",
+          "format": "double",
+          "description": "The numeric value that the extracted text denotes."
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "AgeResolution":{
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseResolution"
+        },
+        {
+          "$ref": "#/definitions/QuantityResolution"
+        }
+      ],
+      "type":"object",
+      "x-ms-discriminator-value": "Age",
+      "properties": {
+        "unit": {
+          "type": "string",
+          "enum": [
+            "Year",
+            "Month",
+            "Week",
+            "Day",
+            "Unspecified"
+          ],
+          "x-ms-enum": {
+            "modelAsString": true
+          },
+          "description": "The Age Unit of measurement"
+        },
+        "duration":{
+          "type":"string",
+          "description": "A value formatted based on the ISO 8601 (https://en.wikipedia.org/wiki/ISO_8601#Durations), or the constant string 'Unspecified' if the age unit is not known."
```

1. Should specifically mention the duration for ISO 8601, which most people only think of as date-time:
```suggestion
          "description": "A value formatted based on the ISO 8601 duration (https://en.wikipedia.org/wiki/ISO_8601#Durations), or the constant string 'Unspecified' if the age unit is not known."
```
2. The REST API archboard is not particular fond of ISO 8601 durations because few languages' base class libraries support them completely. In fact, even in this case, I'm confused why you have an enum that is the unit of measure and an ISO 8601 duration, which dictates the unit of measure like P3Y for 3 years. Seems you shouldn't have both.

Given the archboard's preference, I recommend keeping the unit of measure enum, and just have `duration` be a number.

---

I removed the duration property as it does not provide any added value.